client/blocks/taxonomy-manager/list.jsx

```diff
@@ -47,19 +46,9 @@ export class TaxonomyManagerList extends Component {
 		requestedPages: [ 1 ],
 	};
 
-	UNSAFE_componentWillMount() {
-		this.termIds = map( this.props.terms, 'ID' );
-	}
-
-	UNSAFE_componentWillReceiveProps( newProps ) {
-		if ( newProps.terms !== this.props.terms ) {
-			this.termIds = map( newProps.terms, 'ID' );
-		}
-	}
-
 	getTermChildren( termId ) {
 		const { terms } = this.props;
-		return filter( terms, { parent: termId } );
+		return terms.filter( ( term ) => term.parent === termId );
```

Comment:
Be careful here: looking at the `getTermsForQueryIgnoringPage` selector, it can return `null`. And `defaultProps` will kick in only when the prop is `undefined`?

Reply:
Yup, fixed in https://github.com/Automattic/wp-calypso/pull/57819/commits/05a9a154d5581270a9b43149242b9b18ebd88719 👍🏻